test/components/generators/chat/test_openai.py

```diff
@@ -937,3 +971,113 @@ def test_live_run_with_tools_streaming(self, tools):
         assert tool_call.tool_name == "weather"
         assert tool_call.arguments == {"city": "Paris"}
         assert message.meta["finish_reason"] == "tool_calls"
+
+    def test_openai_chat_generator_with_toolset_initialization(self, tools, monkeypatch):
```

Comment:
Speaking of tests:
- let's try to put most of them in test_toolset.py
- please remove those that seem redundant to you
- remove superfluous comments

Reply:
Regrouped them for now, let me see coverage and then I'll start removing stuff 🙏